src/dwarf/resolver.rs

```diff
@@ -162,39 +228,24 @@ impl DwarfResolver {
             return Ok(elf_r)
         }
 
-        self.ensure_debug_info_syms()?;
-        let dis_ref = self.debug_info_syms.borrow();
-        let debug_info_syms = dis_ref.as_ref().unwrap();
-        let mut idx =
-            match debug_info_syms.binary_search_by_key(&name.to_string(), |v| v.name.to_string()) {
-                Ok(idx) => idx,
-                _ => return Ok(vec![]),
-            };
-        while idx > 0 && debug_info_syms[idx].name.eq(name) {
-            idx -= 1;
-        }
-        if !debug_info_syms[idx].name.eq(name) {
-            idx += 1;
-        }
-        let mut found = vec![];
-        while debug_info_syms[idx].name.eq(name) {
-            let DWSymInfo {
-                addr,
-                size,
-                sym_type,
-                ..
-            } = debug_info_syms[idx];
-            found.push(SymInfo {
-                name: name.to_string(),
-                addr: addr as Addr,
-                size,
-                sym_type,
-                file_offset: 0,
-                obj_file_name: None,
-            });
-            idx += 1;
-        }
-        Ok(found)
+        let mut cache = self.cache.borrow_mut();
+        let () = self.ensure_debug_syms(&mut cache)?;
+        // SANITY: The above `ensure_debug_syms` ensures we have `debug_syms`
```

Comment:
why can't "ensure_debug_syms()" also return debug_syms reference, avoiding these .unwrap()s?

Reply:
It's the same issue issue we have in `ElfParser`:
https://github.com/libbpf/blazesym/blob/739708b8ca4072f958e331f3a4ad2f4061dd492a/src/elf/parser.rs#L228-L231

I think that may be possible once one of the new borrow checkers lands.

Reply:
Ah, right, I already asked this before :)